docs/deployment/deploy_gardenlet_manually.md

```diff
@@ -467,59 +434,168 @@ This helm chart creates:
+
+On reconciliation, gardenlet downloads the Helm chart, renders it with the provided values, and then applies it to its own cluster.
+Hence, in order to keep a gardenlet up-to-date, it is enough to bump the OCI URL for the Helm chart.
+This way, network connectivity to the cluster in which gardenlet runs is not required at all (at least for deployment purposes).
+
+An example `Gardenlet` resource looks like this:
+
+```yaml
+apiVersion: seedmanagement.gardener.cloud/v1alpha1
+kind: Gardenlet
+metadata:
+  name: local
+  namespace: garden
+spec:
+  deployment:
+    replicaCount: 1
+    revisionHistoryLimit: 2
+    helm:
+      ociRepository:
+        ref: localhost:5001/local-skaffold_gardenlet_chart:v1.97.0
+  config:
+    apiVersion: gardenlet.config.gardener.cloud/v1alpha1
+    kind: GardenletConfiguration
+    gardenClientConnection:
+      kubeconfigSecret:
+        name: gardenlet-kubeconfig
+        namespace: garden
+    controllers:
+      shoot:
+        reconcileInMaintenanceOnly: true
+        respectSyncPeriodOverwrite: true
+      shootState:
+        concurrentSyncs: 0
+    featureGates:
+      CoreDNSQueryRewriting: true
+      DefaultSeccompProfile: true
+      HVPA: true
+      HVPAForShootedSeed: true
+      IPv6SingleStack: true
+      ShootManagedIssuer: true
+    etcdConfig:
+      featureGates:
+        UseEtcdWrapper: true
+    logging:
+      enabled: true
+      vali:
+        enabled: true
+      shootNodeLogging:
+        shootPurposes:
+        - infrastructure
+        - production
+        - development
+        - evaluation
+    seedConfig:
+      apiVersion: core.gardener.cloud/v1beta1
+      kind: Seed
+      metadata:
+        labels:
+          base: kind
+      spec:
+        backup:
+          provider: local
+          region: local
+          secretRef:
+            name: backup-local
+            namespace: garden
+        dns:
+          provider:
+            secretRef:
+              name: internal-domain-internal-local-gardener-cloud
+              namespace: garden
+            type: local
+        ingress:
+          controller:
+            kind: nginx
+          domain: ingress.local.seed.local.gardener.cloud
+        networks:
+          nodes: 172.18.0.0/16
+          pods: 10.1.0.0/16
+          services: 10.2.0.0/16
+          shootDefaults:
+            pods: 10.3.0.0/16
+            services: 10.4.0.0/16
+        provider:
+          region: local
+          type: local
+          zones:
+          - "0"
+        settings:
+          excessCapacityReservation:
+            enabled: false
+          scheduling:
+            visible: true
+          verticalPodAutoscaler:
+            enabled: true
+```
+
+When you delete this resource, nothing happens: gardenlet remains running with the configuration as before.
+However, self-upgrades are obviously not possible anymore.
+In order to upgrade it, you have to either recreate the `Gardenlet` object, or deploy the Helm chart.
```

Comment:
Do you talk about the discussed improvement in `gardener-operator`? This is not part of this PR, yet. I will take this into a separate follow-up PR and augment the docu accordingly.

Reply:
No, I was just suggesting to explicitly document how an update would be triggered, i.e. show an extract from the gardenlet resource with the OCI artefact version changed and highlight that this will cause the new version to be rolled out.
From my point of view the documentation states this rather implicitly (`On reconciliation...`).